File: src/commands_of_path.py
#!/usr/bin/env python
import os
import stat
import string

for d in os.environ['PATH'].split(':'):
    try:
        for f in os.listdir(d):
            mode = os.lstat(d + '/' + f)[stat.ST_MODE]
            if not stat.S_ISDIR(mode):
                print(f)
    except FileNotFoundError:
        print(f"Can't open dir {d}: No such file or directory")

File: src/commands_of_path.pl
#!/usr/bin/env perl
use NewPerl;
use English;

foreach my $dir (split(/:/, $ENV{'PATH'})) {
    if (opendir(DIR, $dir)) {
        grep {!/^\./ && print "$_\n" } readdir(DIR);
        closedir DIR;
    } else {
        warn "Can't open dir $dir: $OS_ERROR\n"; } }
